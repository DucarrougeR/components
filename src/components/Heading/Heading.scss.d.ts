export const lensHeading: string;
export const isDisplay1: string;
export const isDisplay2: string;
export const isDisplay3: string;
export const isHeading1: string;
export const isHeading2: string;
export const isHeading3: string;
export const isHeading4: string;
export const isHeading5: string;
export const isHeading6: string;
export const fontWeightLight: string;
export const fontWeightNormal: string;
export const fontWeightSemiBold: string;
export const fontWeightBold: string;
export const fontWeightExtraBold: string;
export const transformUpper: string;
export const transformLower: string;
export const transformCaps: string;
export const transformNone: string;
