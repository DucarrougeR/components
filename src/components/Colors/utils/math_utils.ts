/**
 * Utility to convert a numerical value to a percent given the value and a max value.
 */
export const toPerc = (value: number, maxValue: number) =>
  Math.round((value / maxValue) * 100)
