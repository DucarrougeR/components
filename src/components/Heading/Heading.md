```js
<Heading className="cool" size="d1">Hello</Heading>
<Heading size="d2" level="2">Hello</Heading>
<Heading size="d3" level="3">Hello</Heading>
<Heading size="1" level="1">Hello</Heading>
<Heading size="2" level="2">Hello</Heading>
<Heading size="3" level="3">Hello</Heading>
<Heading size="4" level="4">Hello</Heading>
<Heading size="5" level="5">Hello</Heading>
<Heading level="6" size="">Hello</Heading>
```
