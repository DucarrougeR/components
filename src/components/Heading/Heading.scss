@import '../../styles/util';
@import '../../styles/variables';

.lens-heading {
  margin: 0;
  text-rendering: optimizeLegibility;
  font-family: $brand-font;
  font-weight: $default-font-weight;
}

.is-display-1 {
  font-size: font-size(d1);
  line-height: line-height(d1);
}

.is-display-2 {
  font-size: font-size(d2);
  line-height: line-height(d2);
}

.is-display-3 {
  font-size: font-size(d3);
  line-height: line-height(d3);
}

h1, .is-heading-1 {
  font-size: font-size(1);
  line-height: line-height(1);
}

h2, .is-heading-2 {
  font-size: font-size(2);
  line-height: line-height(2);
}

h3, .is-heading-3 {
  font-size: font-size(3);
  line-height: line-height(3);
}

h4, .is-heading-4 {
  font-size: font-size(4);
  line-height: line-height(4);
}

h5, .is-heading-5 {
  font-size: font-size(5);
  line-height: line-height(5);
}

h6, .is-heading-6 {
  font-size: font-size(6);
  line-height: line-height(6);
}

.font-weight-light {
  font-weight: 300;
}

.font-weight-normal {
  font-weight: 400;
}

.font-weight-semi-bold {
  font-weight: 600;
}

.font-weight-bold {
  font-weight: 700;
}

.font-weight-extra-bold {
  font-weight: 600;
}

.transform-upper {
  text-transform: uppercase;
}

.transform-lower {
  text-transform: lowercase;
}

.transform-caps {
  text-transform: capitalize;
}

.transform-none {
  text-transform: none;
}

