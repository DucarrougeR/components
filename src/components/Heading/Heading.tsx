import * as React from 'react'
import styled, { css } from '../../styled_components'
import { truncate } from '../../styles/typography'
import { lineHeights } from '../../styles/line_heights'
import { fontWeights } from '../../styles/font_weights'
import { RampSizes } from '../../styles/ramp_sizes'

export enum HeadingAlignments {
  Left = 'left',
  Center = 'center',
  Right = 'right'
}

export enum HeadingLevels {
  L1 = '1',
  L2 = '2',
  L3 = '3',
  L4 = '4',
  L5 = '5',
  L6 = '6'
}

export enum HeadingTextTransforms {
  Caps = 'caps',
  Lower = 'lower',
  None = 'none',
  Upper = 'upper'
}

export enum HeadingWeights {
  Bold = 'bold',
  ExtraBold = 'extraBold',
  Light = 'light',
  Normal = 'normal',
  SemiBold = 'semiBold'
}

export interface HeadingGeneratorProps {
  /** Headling tag level mapping for h1-h6 */
  level?: HeadingLevels
}

export interface HeadingProps extends HeadingGeneratorProps {
  /** Size mapping from type ramp */
  size?: RampSizes
  /** Font weight */
  weight?: HeadingWeights
  /** Text tranform  */
  transform?: HeadingTextTransforms
  /** Text align */
  align?: HeadingAlignments
  className?: string
  /** Truncate text on overflow */
  truncate?: boolean
}

const convertHeadingLevelToRampSize = (level: HeadingLevels | undefined) => {
  switch (level) {
    case HeadingLevels.L1:
      return RampSizes.One
    case HeadingLevels.L2:
      return RampSizes.Two
    case HeadingLevels.L3:
      return RampSizes.Three
    case HeadingLevels.L4:
      return RampSizes.Four
    case HeadingLevels.L5:
      return RampSizes.Five
    case HeadingLevels.L6:
      return RampSizes.Six
    default:
      return RampSizes.Three
  }
}

/**
 * Headings are used to help users understand  what a major section of an interface is about, for example the labeling
 * of a page or a title of a card component.
 */
const HeadingGenerator: React.SFC<HeadingProps> = ({ level, ...args }) => {
  // This prevents our props from being passed directly to the underlying h* tags, which ultimately
  // would cause some warnings. Ideally we would define the return type for this function, and
  // Typescript would warn us when passing props that are invalid.
  //
  // See https://reactjs.org/warnings/unknown-prop.html
  const props = Object.assign({}, args)
  delete props.align
  delete props.size
  delete props.transform
  delete props.truncate
  delete props.weight

  switch (level) {
    case HeadingLevels.L1:
      return <h1 {...props}>{props.children}</h1>
    case HeadingLevels.L2:
      return <h2 {...props}>{props.children}</h2>
    case HeadingLevels.L4:
      return <h4 {...props}>{props.children}</h4>
    case HeadingLevels.L5:
      return <h5 {...props}>{props.children}</h5>
    case HeadingLevels.L6:
      return <h6 {...props}>{props.children}</h6>
    case HeadingLevels.L3:
    default:
      return <h3 {...props}>{props.children}</h3>
  }
}

function textTransform(transform: HeadingTextTransforms | undefined) {
  switch (transform) {
    case HeadingTextTransforms.Upper:
      return css`
        text-transform: uppercase;
      `
    case HeadingTextTransforms.Lower:
      return css`
        text-transform: lowercase;
      `
    case HeadingTextTransforms.Caps:
      return css`
        text-transform: capitalize;
      `
    case HeadingTextTransforms.None:
    default:
      return css`
        text-transform: none;
      `
  }
}

function alignment(align: HeadingAlignments | undefined) {
  return css`
    text-align: ${align || HeadingAlignments.Left};
  `
}

export const Heading = styled<HeadingProps>(HeadingGenerator)`
  font-size: ${props =>
    props.theme.fontRamp[
      props.size || convertHeadingLevelToRampSize(props.level)
    ]};
  line-height: ${props =>
    lineHeights[props.size || convertHeadingLevelToRampSize(props.level)]};
  font-weight: ${props => fontWeights[props.weight || HeadingWeights.Normal]};
  ${props => textTransform(props.transform)}
  ${props => alignment(props.align)}
  ${props => truncate(props.truncate || false)}
`
