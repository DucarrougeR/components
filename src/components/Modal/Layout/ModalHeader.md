`ModalHeader` is attached to the top of modal and provides a button to close the containing modal. Uses the `withModal` to consume `ModalContext` and get access to the `close` method.

ModalHeader supports all [`Box`](/#!/Box) properties.

```js
<ModalHeader>
  <Heading weight="semiBold">This is a ModalHeader</Heading>
</ModalHeader>
```