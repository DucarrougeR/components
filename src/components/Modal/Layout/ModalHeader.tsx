import * as React from 'react'
import { Box, BoxProps } from '../../Box'
import { Button } from '../../Button'
import { Icon } from '../../Icon'
import { ModalContextProps, withModal } from '../ModalContext'

export interface ModalHeaderProps
  extends ModalContextProps,
    BoxProps<HTMLDivElement> {
  /**
   * Content that will be placed inside the DialogHeader
   * @required
   */
  children: React.ReactNode
}

const Internal: React.SFC<ModalHeaderProps> = ({
  children,
  closeModal,
  ...props
}) => {
  return (
    <Box is="header" p="large" display="flex" flexDirection="row" {...props}>
      <Box mr="xlarge">{children}</Box>
      <Button
        ml="auto"
        alignSelf="end"
        onClick={closeModal}
        size="large"
        p="none"
        style={{ lineHeight: 1 }}
        variant="transparent"
      >
        <Icon name="Close" />
      </Button>
    </Box>
  )
}

export const ModalHeader = withModal(Internal)
