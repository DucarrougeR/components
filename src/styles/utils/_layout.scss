// Returns a value from the $spacing list
@function spacing($key) {
  @return map-lookup($spacing, $key, '$spacing');
}
