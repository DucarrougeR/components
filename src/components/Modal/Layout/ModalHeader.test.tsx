import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../../test/utils/snapshot'

import { Heading } from '../../Heading'
import { ModalHeader } from './ModalHeader'

test('ModalHeader', () => {
  assertSnapshot(
    <ModalHeader>
      <Heading>The Heading for a Dialog</Heading>
    </ModalHeader>
  )
})
