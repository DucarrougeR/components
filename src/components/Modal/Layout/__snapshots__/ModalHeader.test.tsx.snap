// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 1.25rem;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  margin-right: 2rem;
}

.c3 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.75rem;
}

.c5 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-size: 1.125rem;
  margin-left: auto;
  padding: 0rem;
}

.c6 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1em;
  width: 1em;
}

.c2 {
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-transform: none;
}

.c4 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  border-style: solid;
  border-width: 0.0625rem;
  background: transparent;
  border-color: transparent;
  color: #64518A;
}

.c4:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c4[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c4[disabled]:hover,
.c4[disabled]:active,
.c4[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

.c4:hover,
.c4:focus,
.c4.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c4:active,
.c4.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2 c3"
    >
      The Heading for a Dialog
    </h2>
  </div>
  <button
    className="c4 c5"
    style={
      Object {
        "lineHeight": 1,
      }
    }
  >
    <div
      className="c6"
    >
      <svg
        fill="currentColor"
        height="100%"
        viewBox="0 0 24 24"
        width="100%"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          fill="currentColor"
        />
      </svg>
    </div>
  </button>
</header>
`;
